internal/v2/application/command.go

```diff
@@ -0,0 +1,686 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package application
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/cache"
+	sdkCommon "github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/container"
+	"github.com/edgexfoundry/device-sdk-go/internal/transformer"
+	dsModels "github.com/edgexfoundry/device-sdk-go/pkg/models"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+	edgexErr "github.com/edgexfoundry/go-mod-core-contracts/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type CommandProcessor struct {
+	device         *contract.Device
+	deviceResource *contract.DeviceResource
+	cmd            string
+	params         string
+	dic            *di.Container
+}
+
+func NewCommandProcessor(device *contract.Device, dr *contract.DeviceResource, id string, cmd string, params string, dic *di.Container) *CommandProcessor {
+	return &CommandProcessor{
+		device:         device,
+		deviceResource: dr,
+		cmd:            cmd,
+		params:         params,
+		dic:            dic,
+	}
+}
+
+func CommandHandler(isRead bool, vars map[string]string, body string, id string, dic *di.Container) (event *dsModels.Event, err edgexErr.EdgeX) {
```

Comment:
Why is no a `CommandProcessor` receiver?

Reply:
Some of the fields in `CommandProcessor` are initialized in `CommandHandler` (e.g. device, deviceResource, cmd).  
Also autoevent use CommandHandler as well:
https://github.com/edgexfoundry/device-sdk-go/blob/833c05b782f270c396ee40879762f076aaf9331e/internal/autoevent/executor.go#L93
might be better to remain the same?